src/sortStudents.ts

```diff
@@ -1,16 +1,79 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function checkOrder(num1: number, num2: number, order: SortOrder): number {
+  if (order === 'desc') {
+    return num2 - num1;
+  }
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return num1 - num2;
+}
+
+function getAverageGrade(student: Student): number {
+  const totalGrade = student.grades.reduce(
+    (sum: number, cur: number) => sum + cur, 0,
+  );
+
+  return totalGrade / student.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents: Student[] = [...students];
+
+  // eslint-disable-next-line default-case
+  switch (sortBy) {
+    case SortType.Name:
+      copyStudents.sort(
+        (prev: Student, cur: Student) => prev.name.localeCompare(cur.name),
+      );
+      break;
+    case SortType.Surname:
+      copyStudents.sort(
+        (prev: Student, cur: Student) => {
+          return prev.surname.localeCompare(cur.surname);
+        },
+      );
+      break;
```

Comment:
try to unite cases with similar logic in on case 
```suggestion
    case SortType.Name:
    case SortType.Surname:
      copyStudents.sort(
        (prev: Student, cur: Student) => {
          return prev[sortBy].localeCompare(cur[sortBy]);
        },
      );
      break;
```